jedi/evaluate/gradual/typing.py

```diff
@@ -448,6 +450,27 @@ def py__call__(self, func_context_set):
         return func_context_set
 
 
+class NewTypeFunction(_BaseTypingContext):
+    def py__call__(self, arguments):
+        return ContextSet(
+            NewType(
+                self.evaluator,  # MWC: Not sure what I'm doing here
+                contextualized_node.context,
+                contextualized_node.node,
+                contextualized_node.infer(),  # MWC: Not sure what I'm doing here
```

Comment:
The evaluator part is ok, all contexts have it. The second one, might work for your test case (not even sure), but not all test cases. The contextualized_node here is an arglist node: https://github.com/davidhalter/parso/blob/master/parso/python/grammar37.txt#L124. This is not what you want to be inferred. What we want is probably:

```
...
        ordered_args = arguments.unpack()
        next(ordered_args, None)
        second_arg = next(ordered_args, None)
        if second_arg is None:
            return NO_CONTEXTS
        return ContextSet(
            NewType(
                self.evaluator,
                contextualized_node.context,
                contextualized_node.node,
                second_arg.infer(),
            ) for contextualized_node in arguments.get_calling_nodes())
```